src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/ResultItem.tsx

```diff
@@ -303,24 +309,35 @@ export const ResultItem: React.FC<ResultItemProps & AccessibilityProps> = ({
             <DestinationIcon style={styles.iconContainer} />
             <View style={styles.departureTimes}>
               <ThemeText type="body__tertiary" color="primary" testID="endTime">
-                {formatToClock(tripPattern.expectedEndTime, language, 'ceil')}
+                {(legs.length && isLegFlexibleTransport(legs[legs.length - 1])
```

Comment:
I suggest to check for `legs.length > 0` instead, to avoid the case that legs has length but 0.

Reply:
I don't think there is a difference? https://stackoverflow.com/questions/32911424/array-length-vs-array-length-0